src/com/cdd/bao/validator/RemappingChecker.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * BioAssay Ontology Annotator Tools
+ *
+ * (c) 2014-2018 Collaborative Drug Discovery Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License 2.0
+ * as published by the Free Software Foundation:
+ *
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package com.cdd.bao.validator;
+
+import java.util.*;
+
+/*
+	Use logic in this class to detect cycles in remapping directives. 
+*/
+
+public class RemappingChecker
+{
+	public static void validateRemappings(Map<String, String> remappings) throws RemappingException
+	{
+		List<String> dependencies = new ArrayList<>();
```

Comment:
This needs to be a `LinkedHashSet`. The `add` method of a Set returns true if the element was added and false if not (because it was already an element of the set). List.add probably always allows adding an element. This method needs to have a test.